astropy/coordinates/name_resolve.py

```diff
@@ -9,17 +9,24 @@
 """
 
 # Standard library
+import hashlib
 import os
 import re
+import shelve
+import shutil
 import socket
 import urllib.request
 import urllib.parse
 import urllib.error
+from warnings import warn
 
 # Astropy
 from astropy import units as u
 from .sky_coordinate import SkyCoord
 from astropy.utils import data
+from astropy.utils.data import (_get_download_cache_locs,
```

Comment:
It's really better not to use internal functions like this if there's an atlernative. At a minimum you're moving implementation details outside astropy.utils.data.

Reply:
Yes, this will be impacted by #9182